### ISSUE TITLE
Lazy loading on Group Edit Page

### ISSUE THREAD

**Issue by bananabr** [gophish/gophish#816]
When the number o members in a group is huge (~3000), as the group details/edit page loads all members to memory and leaves pagination and sorting to the frontend, some browsers may freeze or show hangup warnings.

I think that for groups larger than 100 members, some kind of lazy loading must be implemented. The same problem applies to the campaign details page.

**Comment by jordan-wright** [comment]
Yes, this is a known issue. The long term fix is to leverage Datatables' ajax loading capabilities, but this hasn't been implemented yet since I've been working on other feature requests.

I'm happy to review pull requests if you or anyone else would like to send one up.

**Comment by bananabr** [comment]
Altough I have web development experience, I am not familiar with Go and the framework gophish uses. But if you can give me some directions on how to setup a development environment, I can work on this.

**Comment by jordan-wright** [comment]
Sure thing!

As far as setting up a development environment, you'll first need to set up Gophish from source: https://gophish.gitbooks.io/user-guide/content/installation.html

Then, you'll just need to `npm install` to get all the frontend dependencies.

The bad part about this issue is that there's a lot of moving parts. You'll have to implement the ajax handlers (I'd probably make `ajax.go` in the `controllers/` package or I'd consider seeing if I could hook together the API to the format datatables expects and just add parameters like `limit` and `offset` to the API to avoid split brain work with entire Ajax handlers).

Then you'd have to implement the logic in the models package to handle loading the right amount of results at the correct offsets.

Finally, you'd have to implement the front end JS work on the various pages.

I totally understand if such a big feature seems daunting. This is one of those things that I'd be able to get to eventually, but right now I'm finishing up a refactor of how we send emails, then I have a couple of other things in line before I'd be able to get to it.

**Comment by Reedtechno** [comment]
Not sure if I should create a separate issue for this or not, but we are having a similar issue but with the Users & Groups page loading all groups on page load.  My teammate has been developing a tool to sync AD security groups into GoPhish and has done so effectively.  Now we have 1500+ groups and it takes 2.5 minutes for the page to load...

If this should be a separate issue, let me know and I can create one.
